### PR TITLE
Compile Ruby with jemalloc

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

This sets up Ruby with jemalloc instead of malloc for the Docker development setup.

Not only there's a wealth of evidence on the internet, but also in my experience, this single change consistently dropped the memory usage of a Rails app I work on ~10%. I sleep much better since then :sweat_smile: 

#### What should we test?

I tested this myself running it in development with no issue.

#### Release notes

Run docker development setup with Jemalloc
Changelog Category: Technical changes

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Nice. We can do this in production as well, right?

**Comment by Matt-Yorkley** [comment]
Background in case any other devs are interested:

https://www.carmenhchung.com/how-we-halved-memory-with-jemalloc.html
https://www.speedshop.co/2017/12/04/malloc-doubles-ruby-memory.html

**Comment by sauloperez** [comment]
Yes, that was my initial plan

**Review by Matt-Yorkley — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -7,7 +7,23 @@ RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
...
 # Install all the requirements
-RUN apt-get update && apt-get install -y curl git build-essential software-properties-common wget zlib1g-dev libssl1.0-dev libreadline-dev libyaml-dev libffi-dev libxml2-dev libxslt1-dev wait-for-it imagemagick unzip
+RUN apt-get update && apt-get install -y \
+  curl \
+  git \
+  build-essential \
+  software-properties-common \
+  wget \
+  zlib1g-dev \
+  libssl1.0-dev \
```
Comment (Matt-Yorkley) [inline Dockerfile:17]: Hey, whilst you're touching this file; do you want to remove this line as well? :smile:

**Review by andrewpbrett — APPROVED** [review]

**Pull request merged**